src/test/java/br/com/sbs/cubatech/builder/CourseBuilder.java

```diff
@@ -0,0 +1,30 @@
+package br.com.sbs.cubatech.builder;
+
+import br.com.sbs.cubatech.category.Status;
+import br.com.sbs.cubatech.category.SubCategory;
+import br.com.sbs.cubatech.course.Course;
+import br.com.sbs.cubatech.course.CourseVisibility;
+
+public class CourseBuilder {
+
+    private final Course course;
+
+    public CourseBuilder(String name, String urlCode, Integer timeToFinishInHours, String instructor, SubCategory subCategory){
+        this.course = new Course(name, urlCode, timeToFinishInHours, instructor, subCategory );
+    }
+
+    public CourseBuilder isPrivate(){
+        this.course.setCourseVisibility(CourseVisibility.PRIVATE);
+        return this;
+    }
+
+    public CourseBuilder isPublic(){
+        this.course.setCourseVisibility(CourseVisibility.PUBLIC);
+        return this;
+    }
```

Comment:
Os nomes `isPrivate` e `isPublic` dão a entender que são consultas, perguntado se o objeto é ou não privado/público.

Mas na verdade isso é um comando, mandando o objeto tornar-se privado/público.

Podia ser:

```suggestion
    public CourseBuilder private(){
        this.course.setCourseVisibility(CourseVisibility.PRIVATE);
        return this;
    }

    public CourseBuilder public(){
        this.course.setCourseVisibility(CourseVisibility.PUBLIC);
        return this;
    }
```